libs/langchain/tests/integration_tests/embeddings/test_openai.py

```diff
@@ -71,11 +71,51 @@ def test_openai_embedding_with_empty_string() -> None:
     assert len(output[1]) == 1536
 
 
+@pytest.mark.asyncio
+async def test_sync_async_embed_documents_equal() -> None:
+    documents = ["foo bar", "bar foo", "foo"]
+    embedding = OpenAIEmbeddings()
+    sync_output = embedding.embed_documents(documents)
+    async_output = await embedding.aembed_documents(documents)
+    assert np.isclose(sync_output, async_output, atol=1e-4).mean() > 0.95
+
+
+@pytest.mark.asyncio
+async def test_sync_async_embed_query_equal() -> None:
+    query = "foo bar"
+    embedding = OpenAIEmbeddings()
+    sync_output = embedding.embed_query(query)
+    async_output = await embedding.aembed_query(query)
+    assert np.isclose(sync_output, async_output, atol=1e-4).mean() > 0.95
+
+
+@pytest.mark.asyncio
+async def test_with_without_tokenizing_equal() -> None:
+    documents = ["foo bar baz bum buzz boom"]
+    with_tokenizing = OpenAIEmbeddings()
+    without_tokenizing = OpenAIEmbeddings(embedding_ctx_length=None)
+    tokenize_output = with_tokenizing.embed_documents(documents)
+    text_output = without_tokenizing.embed_documents(documents)
+    assert np.isclose(tokenize_output, text_output, atol=1e-4).mean() > 0.95
```

Comment:
this actually rarely passes, can't tell if it's bc tokenizing changes behavior or because of a bug (doesn't seem to be just randomness bc other tests don't fail as much)

Reply:
this is also failing for me

Reply:
@hwchase17 @baskaryan please see my comment below https://github.com/langchain-ai/langchain/pull/7815/#issuecomment-1685416343